src/rviz/robot/robot_link.cpp

```diff
@@ -620,49 +605,22 @@ void RobotLink::createEntityForGeometryElement(const urdf::LinkConstSharedPtr& l
     offset_node->setPosition(offset_position);
     offset_node->setOrientation(offset_orientation);
 
-    static int count = 0;
-    if (default_material_name_.empty())
+    if (default_material_.isNull() || material)
     {
-      default_material_ = getMaterialForLink(link);
-
-      std::stringstream ss;
-      ss << default_material_->getName() << count++ << "Robot";
-      std::string cloned_name = ss.str();
-
-      default_material_ = default_material_->clone(cloned_name);
-      default_material_name_ = default_material_->getName();
+      // latest used material becomes the default for this link
+      default_material_ = getMaterialForLink(link, material);
     }
 
     for (uint32_t i = 0; i < entity->getNumSubEntities(); ++i)
     {
-      default_material_ = getMaterialForLink(link, material_name);
-      std::stringstream ss;
-      ss << default_material_->getName() << count++ << "Robot";
-      std::string cloned_name = ss.str();
-
-      default_material_ = default_material_->clone(cloned_name);
-      default_material_name_ = default_material_->getName();
-
       // Assign materials only if the submesh does not have one already
-
       Ogre::SubEntity* sub = entity->getSubEntity(i);
       const std::string& material_name = sub->getMaterialName();
 
       if (material_name == "BaseWhite" || material_name == "BaseWhiteNoLighting")
       {
-        sub->setMaterialName(default_material_name_);
+        sub->setMaterial(default_material_);
       }
-      else
-      {
-        // Need to clone here due to how selection works.  Once selection id is done per object and not per material,
```

Comment:
@wjwwood I'm not sure whether this code is already obsolete, because selection handling was changed.
Anyway, removing this code, I didn't observed any issues in selecting links. However, probably you better know, what to test...